src/plugins/resolver/resolver.c

```diff
@@ -73,6 +73,9 @@ static void resolverInit (resolverHandle * p, const char * path)
 	p->tempfile = 0;
 
 	p->path = path;
+
+	p->uid = getuid ();
+	p->gid = getgid ();
```

Comment:
does it really matter if chown fails ? if it succeeds, the permissions are restored, if not, we most likely don't have the rights to change/restore them.